File: java/2965.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] g) {
        int n = g.length, rep = -1;
        boolean f[] = new boolean[n * n + 1];
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                if (f[g[i][j]])
                    rep = g[i][j];
                else
                    f[g[i][j]] = true;
        for (int i = n * n; i >= 0; i--)
            if (!f[i])
                return new int[] { rep, i };
        return null;
    }
}

File: python3/2965.py
class Solution:
    def findMissingAndRepeatedValues(self, g: List[List[int]]) -> List[int]:
        n = len(g)
        rep = -1
        f = [True] * (n * n + 1)
        for i in g:
            for j in i:
                if f[j]:
                    f[j] = False
                else:
                    rep = j
        for i in range(1, n * n + 1):
            if f[i]:
                return [rep, i]
        return None
